CHANGELOG.md

```diff
@@ -4,6 +4,13 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.43.0]
+
+## What's Changed
+* Update the rust toolchain to `nightly-2023-12-14` by @tautschnig and @adpaco-aws
```

Comment:
```suggestion
### What's Changed
* Rust toolchain upgraded to `nightly-2023-12-14` by @tautschnig and @adpaco-aws
```

Reply:
Implemented as suggested.

Reply:
No, there should be another `#` in `## What's Changed`.

Reply:
Got it!